src/controllers/users.controllers.js

```diff
@@ -10,13 +26,92 @@ export default class Users {
    * @param {Object} req valiable
    * @param {Object} res valiable
    * @return {Object} res
-  */
+   */
   static async signup(req, res) {
     try {
       const { data } = await User.register(req.body);
-      return res.status(200).json({ message: 'user created successful', data });
+      const userToken = Jwt.generateToken({ data }, '1h');
+      if (userToken) {
+        data.email_token = userToken;
+      }
+      await data.save();
+      const verificationEmail = verifyEmailTemplate(userToken);
+      sendEmail(
+        emailConfig({
+          email: data.email,
+          subject: 'Brogrammers email verification',
+          content: verificationEmail,
+        })
+      );
+      return res.status(201).json({
+        message: 'Check your email to verify your account',
+        user: data,
+      });
     } catch (e) {
-      return res.status(500).json(e);
+      return res.status(500).json({
+        error: e.message,
+        message: 'Failed to register a new user',
+      });
+    }
+  }
+
+  static async verifyEmail(req, res) {
+    try {
+      const emailToken = req.params.token;
+      Users.verifyUser(emailToken, res);
+    } catch (error) {
+      res.status(500).json({ 'Error:': error });
+    }
+  }
+
+  static async verifyUser(emailToken, res) {
+    const user = await User.findByEmailToken(emailToken);
+    if (!user) {
+      return res.status(404).json({
+        message: 'User not found',
+      });
+    }
+    user.email_token = null;
+    user.verified = true;
+    await user.save();
+    res.status(200).send({ message: 'Your account has been verified successfully!', verified: true });
+    console.log('User Verified Successfully');
```

Comment:
remove `console.log()`,It is just for debugging purposes only ,no need to ship it on staging

Reply:
Done